.claude/scripts/pipefail-grep-guard.sh

```
@@ -0,0 +1,1366 @@
+# Single quotes honour no escapes; double quotes and code do. A `#` that starts a
+# comment ends the line's quote significance, so an apostrophe in prose
+# (`# don't`) cannot leak an open quote into the following lines.
+_QUOTE_STATE=0
+advance_quote_state() {
+  local s="$1" st="$2" i c prev=''
+  for ((i = 0; i < ${#s}; i++)); do
+    c="${s:i:1}"
+    case "$st" in
+      0)
+        case "$c" in
+          '\') ((i++)) ;;
```

Shellcheck (0.11.0)

[info] Want to escape a single quote? echo 'This is how it'\''s done'.

(SC1003)